.github/scripts/promote_branch.sh

```diff
@@ -156,7 +153,9 @@ fi
 
 git checkout $SOURCE_BRANCH
 git push origin $SOURCE_BRANCH:$TARGET_BRANCH
-git push origin $SOURCE_BRANCH:$SRE_DUPLICATE_BRANCH
+if [ ! -z "$SRE_DUPLICATE_BRANCH" ] ; then
```

Comment:
I was gonna say use `-n` instead of the negation of `-z`. But I also noticed that this variable will not be set at all otherwise. I advocate for `set -eu` as a default. With that, it would fail on "variable undefined". We don't have that here, so it would work. But I'd say either do `SRE_DUPLICATE_BRANCH=` above line 98, or do `[ -v SRE_DUPLICATE_BRANCH ]` (without dollar sign) here to just check if it's set.

Reply:
I don't want the script to fail. `[ -v SRE_DUPLICATE_BRANCH ]` and `[ ! -z $SRE_DUPLICATE_BRANCH ]` are behaving identically in my tests (true if it is set to a string, false otherwise). Why change it?

Reply:
Let me try to rephrase it. It seems wrong to check the value of a variable that might not exists (it works, but only because this script does not have `set -u`). So that's why I would suggest either `test -v` or assigning an empty string to `SOMETHING_ELSE` before anything else. If you don't take this as a valid point, fair enough.

In that case I would still suggest to use `test -n` instead of `test ! -z`. It's equivalent, but it's similar to using `test ! $num -gt 0` instead of `test $num -le 0` for example - why use a negative test when there is an operator for just what you need? It's definitely nitpicking for sure, so consider this optional.

Reply:
Thanks for the explanation. Updated to `-v`

Reply:
ok, thanks